discord/src/tools/searchMessages.ts

```diff
@@ -0,0 +1,175 @@
+import { TextChannel, Attachment, Embed, Message, Collection } from 'discord.js';
+import { client } from '../client.js';
+import { toRFC3339, createDataset } from '../utils.js';
+
+export async function searchMessages() {
+  if (!process.env.QUERY || !process.env.LIMIT) {
+    throw new Error('QUERY and LIMIT environment variables are required for searchMessages');
+  }
+
+  const query = process.env.QUERY;
+  const limit = parseInt(process.env.LIMIT, 10);
+  const guildId = process.env.GUILDID; // Optional
+  const channelId = process.env.CHANNELID; // Optional
+
+  // If channelId is provided, guildId must also be provided
+  if (channelId && !guildId) {
+    throw new Error('GUILDID must be provided when CHANNELID is specified');
+  }
+
+  const results = [];
+
+  if (guildId && channelId) {
+    // Search in specific channel
+    const guild = await client.guilds.fetch(guildId);
+    const channel = await guild.channels.fetch(channelId);
+    
+    if (!(channel instanceof TextChannel)) {
+      throw new Error('Specified channel is not a text channel');
+    }
+
+    try {
+      const messages = await channel.messages.fetch({ 
+        limit: Math.min(100, limit)
```

Comment:
should we fetch more than just 100 messages to search? or is it too expensive/slow/rate-limit concern..?

Reply:
I decided to nuke this tool because it was bad